src/main.rs

```diff
@@ -0,0 +1,71 @@
+use anyhow::Result;
+use clap::Parser;
+use std::{fs, path, process};
+
+#[derive(Parser)]
+struct Cli {
+    #[arg(short = 'f', long = "file")]
+    file: path::PathBuf,
+}
+
+const RUSTTEST_ERROR: &str = "//~^ ERROR ";
+const DG_ERROR: &str = "// { dg-error \"";
+
+fn main() -> Result<()> {
+    if let Err(error) = try_parse() {
+        eprintln!("Error: {:?}", error);
+        process::exit(1);
+    }
+    Ok(())
```

Comment:
so is it not possible to simplify the code here just by calling `try_parse()` and letting it return? does it produce an error?

Reply:
Yes, it produces error

---

![image](https://github.com/Rust-GCC/rusttest-to-dg/assets/95980383/775c41e9-2c2e-4d07-8fae-3dbf01215d04)

Reply:
Are you referring to this? Yes, it does compile :P
```suggestion
    try_parse()
```

Reply:
yes, I was referring to the second snippet you posted :) I think it's cleaner to let anyhow handle all of that for us. can you commit that and then we'll merge it?

Reply:
It's Updated now:)

I have been following this approach: https://github.com/chmln/sd/blob/master/src/main.rs#L22-L29. It only uses `main` for returning `anyhow` error code and `try_main` will handle all other things.